### PR TITLE
Add documentation for Proxmox

### PR DESCRIPTION
**Pull request by maximiliankolb**
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (planned Satellite 6.13)
* [x] Foreman 3.4/Katello 4.6 (EL8 only)
* [x] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only)

### CODE REVIEW

File: guides/common/modules/proc_installing-the-proxmox-plug-in.adoc
Comment (ekohl) [inline guides/common/modules/proc_installing-the-proxmox-plug-in.adoc:12]: You don't need to specify `rubygem-fog-proxmox`, since the plugin itself already depends on it. Then it should also use the `{package-install}` macro:
```suggestion
# {package-install} rubygem-foreman_fog_proxmox
```

Though I'd really like to avoid this and instead rely on the installer.

**Review by ekohl — COMMENTED** [review]
Given the various WIP parts, should it be a draft?

I've also been hesitant to claim support for it, given there is only a single developer on it right now who does it in his spare time (see https://github.com/orgs/theforeman/teams/proxmox). Adding installer support for it should be fairly easy though.

**Comment by maximiliankolb** [comment]
@ekohl I've added the comments to help remember to expand the docs later on. Yes, they are incomplete, but nevertheless helpful for basic workflows. CLI commands are also missing and might get added at a later stage.

My colleagues are actively working on Proxmox with Foreman 3.3+. Would you prefer to only show this for orcharhino then?

**Comment by ekohl** [comment]
@maximiliankolb Not strictly, though it should be hidden for Satellite given it doesn't ship it. However, if you're working on it then it would be best to also have the permissions to actually make changes. In practice that may mean adding @Manisha15 to the GH team. And also open PRs to the installer. An equivalent of https://github.com/theforeman/puppet-foreman/commit/1b5eec23169a585f506a8317823c8bb41a6224ff would be a good start. From there I can also guide you through the installer itself.

File: guides/common/modules/proc_installing-the-proxmox-plug-in.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_installing-the-proxmox-plug-in.adoc:12]: I've applied this change using "package-install-project". I will also have a look at the link below on how to get Proxmox added to foreman-installer.

**Comment by Manisha15** [comment]
> @maximiliankolb Not strictly, though it should be hidden for Satellite given it doesn't ship it. However, if you're working on it then it would be best to also have the permissions to actually make changes. In practice that may mean adding @Manisha15 to the GH team. And also open PRs to the installer. An equivalent of [theforeman/puppet-foreman@1b5eec2](https://github.com/theforeman/puppet-foreman/commit/1b5eec23169a585f506a8317823c8bb41a6224ff) would be a good start. From there I can also guide you through the installer itself.

Hey @ekohl , I am actively working on Proxmox from Atix side. I have opened some issues and PR as well (foreman_fog_proxmox), and will be opening some more PR in near future. It would be great if I can get permission to make changes there.

**Comment by mjivraja** [comment]
Can we add `Proxmox` as an attribute?

**Comment by mjivraja** [comment]
> Sure, what's the usecase? Do you want branding for certain build targets?

If Proxmox is similar to {oVirt} for upstream builds, we can add it. But, I don't have much knowledge about it.

**Comment by ekohl** [comment]
The reason we have `{oVirt}` is branding. Red Hat has a branded product called RHV which is a commercially supported variant of oVirt, similar to Satellite and Foreman (+ Katello). I don't foresee a branded version of Proxmox. Another reason could be like RHEL where you want it consistently expanded including non-breaking spaces but that doesn't apply here either. Adding an attribute doesn't add any value as far as I can see.

**Comment by maximiliankolb** [comment]
> > Sure, what's the usecase? Do you want branding for certain build targets?
> 
> If Proxmox is similar to {oVirt} for upstream builds, we can add it. But, I don't have much knowledge about it.

I am not sure. Can you check if Satellite officially supports Proxmox? I assume no, and I assume there is no downstream product by Red Hat based on Proxmox.

**Comment by mjivraja** [comment]
Ewoud's view is insightful and I definitely agree with the same. :+1:  

Proxmox is not supported at present in Satellite as far as I know.

**Comment by ekohl** [comment]
I've added @Manisha15 to @theforeman/proxmox now.

**Comment by maximiliankolb** [comment]
I am currently also trying to get Proxmox into foreman-installer. However, this may take some more time and not be immediately available of all Foreman releases (if any). So I think we should review and merge this as is, and then improve the install section later on.

Kindly asking for a review.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (adamlazik1) [inline guides/common/modules/proc_creating-hosts-on-proxmox.adoc:44]:
```suggestion
ifndef::satellite[]
```

File: guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc
Comment (adamlazik1) [inline guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc:7]:
```suggestion
. In the {ProjectWebUI}, navigate to *Infrastructure* > *Compute Resources*.
```

File: guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc
Comment (adamlazik1) [inline guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc:7]:
```suggestion
. In the {ProjectWebUI}, navigate to *Infrastructure* > *Compute Resources*.
```

File: guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc
Comment (adamlazik1) [inline guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc:9]: Should it be explained what should be clicked under what circumstances?

File: guides/common/modules/proc_installing-the-proxmox-plug-in.adoc
Comment (adamlazik1) [inline guides/common/modules/proc_installing-the-proxmox-plug-in.adoc:4]:
```suggestion
Install the Proxmox plug-in to attach a Proxmox compute resource provider to {Project}.
```

File: guides/common/modules/proc_installing-the-proxmox-plug-in.adoc
Comment (adamlazik1) [inline guides/common/modules/proc_installing-the-proxmox-plug-in.adoc:13]: Please split.

File: guides/common/modules/proc_installing-the-proxmox-plug-in.adoc
Comment (adamlazik1) [inline guides/common/modules/proc_installing-the-proxmox-plug-in.adoc:15]:
```suggestion
. Optional: In the {ProjectWebUI}, navigate to *Administer* > *About* and select the *Compute Resources* tab to verify the installation of the Proxmox plug-in.
```

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc:15]: Technically yes, practically, I'd say there's a chance we'll leave it as is for a while.

Proxmox *does* support it, but I'm unsure if it's a priority for downstream to have it working/documented.

That why I've left small adoc comments to indicate where one could extend it. Is that OK for you?

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc]: Applied.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc:24]: It's a hint for "future me" to potentially extend the docs.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc]: Applied.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc]: Applied.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc]: Applied.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc]: Applied.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc]: Applied.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc:37]: Yes, it's a TODO for future me.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc]: Applied.

File: guides/common/modules/proc_adding-proxmox-images-to-server.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-images-to-server.adoc]: Slightly reworded.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-hosts-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-hosts-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-hosts-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-hosts-on-proxmox.adoc]: Removed.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-hosts-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-hosts-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-hosts-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-hosts-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_creating-hosts-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-hosts-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc]: Applied and also change the file name.

File: guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_deleting-a-vm-on-proxmox.adoc]: Applied.

File: guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc]: Applied.

File: guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc:9]: Ideally yes, but it's copied from VMware which does not explain it either. I think that's something for a future improvement. :)

File: guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_importing-a-virtual-machine-from-proxmox-into-project.adoc]: Applied.

File: guides/common/modules/proc_installing-the-proxmox-plug-in.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_installing-the-proxmox-plug-in.adoc]: Applied.

File: guides/common/modules/proc_installing-the-proxmox-plug-in.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_installing-the-proxmox-plug-in.adoc]: Applied.

File: guides/common/modules/proc_installing-the-proxmox-plug-in.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_installing-the-proxmox-plug-in.adoc]: Applied.

**Review by maximiliankolb — COMMENTED** [review]
Thank you @adamlazik1 for the review! I've applied almost all your suggestions. I've left some thread open for further discussion/approval.

Some comments still hint towards future improvements of the Proxmox docs.

**Comment by ekohl** [comment]
> I am currently also trying to get Proxmox into foreman-installer. However, this may take some more time and not be immediately available of all Foreman releases (if any). So I think we should review and merge this as is, and then improve the install section later on.

I agree, but it is good we're starting on the proper long term fixes. Too often have we lived with workarounds for ages, and that always bugs me.

File: guides/common/modules/proc_deleting-a-virtual-machine-on-proxmox.adoc
Comment (goarsna) [inline guides/common/modules/proc_deleting-a-virtual-machine-on-proxmox.adoc:11]:
```suggestion
. Optional: If you want to delete the orphaned host, navigate to *Hosts* > *All Hosts* and delete the host manually.
```

File: guides/common/modules/proc_adding-a-proxmox-connection-to-server.adoc
Comment (goarsna) [inline guides/common/modules/proc_adding-a-proxmox-connection-to-server.adoc:7]:
```suggestion
* Ensure that the host and network based firewalls are configured to allow communication from {Project} to Proxmox on TCP port 443.
```

**Review by goarsna — CHANGES_REQUESTED** [review]
I found some small improvements. Despite those the whole chapter IMO looks good.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (adamlazik1) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc:15]: I see. yes, that works. Thanks for explaining.

**Comment by adamlazik1** [comment]
@maximiliankolb LGTM. My only other suggestions are those proposed by @goarsna.

**Comment by mjivraja** [comment]
> @maximiliankolb LGTM. My only other suggestions are those proposed by @goarsna.

I am seeing mine still `Pending` if I am not mistaken.

**Comment by adamlazik1** [comment]
I thought that was resolved in https://github.com/theforeman/foreman-documentation/pull/2142#issuecomment-1508433447?

File: guides/common/modules/proc_adding-a-proxmox-connection-to-server.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-a-proxmox-connection-to-server.adoc:7]: No change; `git grep` tells me it's "network-based".

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc:21]: Applied.

File: guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-proxmox-details-to-a-compute-profile.adoc:23]: Applied.

File: guides/common/modules/proc_deleting-a-virtual-machine-on-proxmox.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_deleting-a-virtual-machine-on-proxmox.adoc:11]: Applied.

**Comment by maximiliankolb** [comment]
Applied all feedback and rebased to master. Will merge if GHA are green & change the installation instructions as soon as I've made progress with adding Proxmox to `foreman-installer`.

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: :heavy_check_mark: 
   1bd6eb7da..01041412d  3.6 -> 3.6
   f89568afe..4e1516f8d  3.5 -> 3.5
   cc9856ec2..10d9e80c4  3.4 -> 3.4
   df33ae576..9819451e2  3.3 -> 3.3

Again, thanks to everyone for their input!

**Pull request merged**
